app/src/main/java/org/mozilla/tv/firefox/settings/SettingsFragment.kt

```diff
@@ -34,11 +35,15 @@ class SettingsFragment : Fragment() {
         val context = view.context
 
         ic_lock.setImageResource(R.drawable.mozac_ic_lock)
-        telemetryButton.isChecked = DataUploadPreference.isEnabled(context)
+        val factory = context.serviceLocator.viewModelFactory
+        val settingsViewModel = ViewModelProviders.of(this, factory).get(SettingsViewModel::class.java)
+        settingsViewModel.dataCollectionEnabled.observe(viewLifecycleOwner, Observer<Boolean> { state ->
+            if (state != null) {
```

Comment:
Optional nit: I personally prefer shorting using `state ?: return` because it flattens out some nesting, but I don't feel strongly about it.